### ISSUE TITLE
Ridge (alpha > 0) penalises the intercept added by add_intercept=True

### ISSUE THREAD

**Issue by ancri** [azmyrajab/polars_ols#49]
### Summary

When `alpha > 0`, the intercept column added by `add_intercept=True` is included in the L2 penalty. This differs from `sklearn.linear_model.Ridge(fit_intercept=True)`, which centres the data and leaves the intercept unpenalised — the usual convention for penalised regression.

The effect is not subtle: the intercept is shrunk toward zero, and because the fit must compensate, the slope estimates are dragged with it. Users who reach for `alpha` as light numerical stabilisation on data with a non-zero mean will get materially biased coefficients with no warning.

### Reproducer

```python
import numpy as np, polars as pl, polars_ols  # noqa: F401
from sklearn.linear_model import Ridge

n = 200
rng = np.random.default_rng(2)
x = rng.normal(size=n)
y = 50.0 + 2.0 * x + rng.normal(size=n)       # true intercept = 50
df = pl.DataFrame({"y": y, "x": x})
Xc = np.column_stack([x, np.ones(n)])

for alpha in (0.0, 1.0, 100.0):
    p = np.array(df.select(pl.col("y").least_squares.ols(
        "x", add_intercept=True, alpha=alpha,
        mode="coefficients").alias("c"))["c"].struct.unnest().row(0))
    rT = Ridge(alpha=alpha, fit_intercept=True).fit(x[:, None], y)
    skF = Ridge(alpha=alpha, fit_intercept=False).fit(Xc, y).coef_
    print(f"alpha={alpha:<7} polars_ols={np.round(p,4)}  "
          f"Ridge(fit_intercept=True)={np.round([rT.coef_[0], rT.intercept_],4)}  "
          f"Ridge(fit_intercept=False)={np.round(skF,4)}")
```

Output:

```
alpha=0.0     polars_ols=[ 2.0185 49.9547]  Ridge(fit_intercept=True)=[ 2.0185 49.9547]  Ridge(fit_intercept=False)=[ 2.0185 49.9547]
alpha=1.0     polars_ols=[ 2.0055 49.706 ]  Ridge(fit_intercept=True)=[ 2.0076 49.9546]  Ridge(fit_intercept=False)=[ 2.0055 49.706 ]
alpha=100.0   polars_ols=[ 1.2201 33.299 ]  Ridge(fit_intercept=True)=[ 1.3109 49.9492]  Ridge(fit_intercept=False)=[ 1.2201 33.299 ]
```

`polars_ols` tracks `Ridge(fit_intercept=False)` exactly, confirming the constant column sits inside the penalty. At `alpha=100` the intercept has moved from its true value of 50 to 33.3, and the slope from 2.0 to 1.22.

I verified the coefficients otherwise solve the textbook weighted-ridge normal equation `(X'WX + αI)⁻¹X'Wy` to ~1e-12, and match `Ridge(fit_intercept=False)` on `[X, 1]` to ~1e-13 — so this is purely about which columns the penalty covers, not about the solver.

### Why it matters more with `sample_weights`

`add_intercept=True` appends the constant column in `_pre_process_data` *before* the `sqrt(w)` rescaling (`least_squares.py` L184-195), so the intercept column becomes `√wᵢ` and its contribution to `X'X` is `Σwᵢ`. With weights that are normalised to sum to 1, `Σw` is tiny relative to a fixed `alpha`, so the intercept is very heavily penalised. In my case `alpha=0.001` with weights summing to 0.068 shrank an intercept whose unpenalised value was 10.4 down to 0.27.

### Suggested resolution

Either of:

1. Exclude the intercept column from the penalty (the sklearn/statsmodels convention) — e.g. by zeroing the corresponding diagonal entry of the `λI` term, or by centring `X` and `y` and recovering the intercept afterward.
2. Keep current behaviour but document it clearly on `OLSKwargs.alpha` / `add_intercept`, since it is surprising relative to the obvious reference implementations.

I'd lean toward (1) as the default with an opt-out flag, but (2) would at least stop people being caught out.

### Workaround for anyone hitting this

Weighted-demean the target and features within group, fit with `add_intercept=False`, then recover `const = ȳ_w − x̄_w·β`. I checked this reproduces `Ridge(fit_intercept=True, sample_weight=w)` to ~1e-15.

### Environment

- `polars_ols` 0.3.5
- `polars` 1.38.1
- Python 3.10.11, Linux x86-64
